starknet_py/devnet_utils/devnet_client.py

```diff
@@ -0,0 +1,309 @@
+        """
+        Create a new block.
+        """
+
+        res = await self._devnet_client.call(method_name="createBlock")
+
+        return res["block_hash"]
+
+    async def abort_block(self, starting_block_hash: Hash) -> List[str]:
+        """
+        This functionality allows simulating block abortion that can occur on mainnet.
+        It is supported in the `--state-archive-capacity full` mode.
+
+        :param starting_block_hash: The state of Devnet will be reverted to the state before `starting_block_hash`.
+        """
+
+        res = await self._devnet_client.call(
+            method_name="abortBlocks",
+            params={"starting_block_hash": _to_rpc_felt(starting_block_hash)},
+        )
+
+        return res["aborted"]
+
+    async def dump(self, path: str):
+        """
+        Dump the state of the devnet to a file.
+        Dumping on request requires providing `--dump-on` mode on the startup.
+
+        :param path: Path to the file.
+        """
+
+        await self._devnet_client.call(
+            method_name="dump",
+            params={"path": path},
+        )
+
+    async def load(self, path: str):
+        """
+        Load the state of the devnet from a file.
+
+        :param path: Path to the file.
+        """
+
+        await self._devnet_client.call(
+            method_name="load",
+            params={"path": path},
+        )
+
+    async def restart(self):
+        """
+        Restart the devnet.
+        """
+
+        await self._devnet_client.call(
+            method_name="restart",
+        )
+
+    async def postman_load(
+        self, network_url: str, address: Optional[str] = None
+    ) -> str:
+        params = {"network_url": network_url}
+        if address is not None:
+            params["address"] = address
+
+        res = await self._devnet_client.call(
+            method_name="postmanLoad",
+            params=params,
+        )
+
+        return res["messaging_contract_address"]
+
+    async def postman_flush(
+        self, dry_run: Optional[bool] = False
+    ) -> PostmanFlushResponse:
+        res = await self._devnet_client.call(
+            method_name="postmanFlush",
+            params={"dry_run": dry_run},
+        )
+        return cast(PostmanFlushResponse, PostmanFlushResponseSchema().load(res))
+
+    # pylint: disable=too-many-arguments
+    async def send_message_to_l2(
+        self,
+        l2_contract_address: Hash,
+        entry_point_selector: Hash,
+        l1_contract_address: Hash,
+        payload: List[Hash],
+        nonce: Hash,
+        paid_fee_on_l1: Hash,
+    ) -> str:
+        res = await self._devnet_client.call(
+            method_name="postmanSendMessageToL2",
+            params={
+                "l2_contract_address": _to_rpc_felt(l2_contract_address),
+                "entry_point_selector": _to_rpc_felt(entry_point_selector),
+                "l1_contract_address": _to_rpc_felt(l1_contract_address),
```

Comment:
To be more precise we should probably use a different method here as L1 addresses are a maximum of 40 characters long, and in `_to_rpc_felt` we allow up to 63 characters. Although it doesn't break anything, I think it can stay as it is for now.